jfrog-cli/artifactory/utils/utils.go

```diff
@@ -158,9 +161,111 @@ func RunCmd(config CmdConfig) error {
 	return nil
 }
 
+/*
+Executes the command and captures the output.
+Analyze each line to match the provided regex.
+*/
+func RunCmdWithOutputParser(config CmdConfig, regExpStruct ...RegExpStruct) error {
+	for k, v := range config.GetEnv() {
+		os.Setenv(k, v)
+	}
+
+	cmd := config.GetCmd()
+	cmdReader, err := cmd.StderrPipe()
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	defer cmdReader.Close()
+	scanner := bufio.NewScanner(cmdReader)
+
+	err = cmd.Start()
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		for _, regExp := range regExpStruct {
+			if regExp.RegExp.FindString(line) != "" {
+				line, err = regExp.ExecFunc(line)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		log.Output(line)
+	}
+	if scanner.Err() != nil {
+		return errorutils.CheckError(scanner.Err())
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	return nil
+}
+
 type CmdConfig interface {
 	GetCmd() *exec.Cmd
 	GetEnv() map[string]string
 	GetStdWriter() io.WriteCloser
 	GetErrWriter() io.WriteCloser
 }
+
+func GetRegExp(regex string) (*regexp.Regexp, error) {
+	excludePathsRegExp, err := regexp.Compile(regex)
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+
+	return excludePathsRegExp, nil
+}
+
+/*
+Mask the credentials information from the line. The credentials are build as user:password
+For example: http://user:password@127.0.0.1:8081/artifactory/path/to/repo
+*/
+func MaskCredentials(line string) (string, error) {
```

Comment:
Let's see fi we can come up with a more generic implementation, which can do the masking according to a regular expression.